Initial release

twolfson opened the pull request:
This is the initial release of `httpretty-fixtures`. Please see documentation for explanation.

I should note that I finished this while intoxicated and will re-review before landing. I am choosing @brettlangdon as my reviewer because I want to verify that the API seems Pythonic/approachable.

In this PR:
- Added library
- Added tests
- Added documentation

/cc @brettlangdon

twolfson commented:
Before landing this PR, we should update the documentation to list out the expected function signature for requests (both parameters and return).

brettlangdon commented on README.rst:
```diff
@@ -13,16 +26,179 @@ Install the module with: ``pip install httpretty_fixtures``
+
+httpretty_fixtures.requests()
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Alias to access all request received by ``HTTPretty``.
```
`s/request/requests/`

twolfson replied:
:+1:

brettlangdon commented on README.rst:
```diff
@@ -13,16 +26,179 @@ Install the module with: ``pip install httpretty_fixtures``
+    # Define our tests
+    class MyTestCase(unittest.TestCase):
+        @CounterServer.run(['counter'])
+        def test_counter_state(self):
```
for this example and the test case below as well, it might be worth adding a second test case to show that the manager state is only maintained on a per test (start/stop) basis.

``` python
@CounterServer.run(['counter'])
def test_counter_state(self):
    # make request
    self.asserEqual(res.text, '1')

@CounterServer.run(['counter'])
def test_counter_state_again(self):
    # make request
    self.assertEqual(res.text, '1')
```

twolfson replied:
Sure, that seems fair

brettlangdon commented:
for the docs should we use just use the `readthedocs` here? you have decent doc strings for everything, should work fine.

brettlangdon commented:
just a few comments, otherwise lgtm to me.

twolfson commented:
With respect to readthedocs, I prefer to write them by hand. I get to go into more detail and provide more examples without cluttering up the source code.

twolfson commented:
I have taken care of your comments, added documentation for the function signature, fixed up some minor errors, and added tests that verify we preserve and don't share state.

brettlangdon commented:
:+1: 

![image](https://cloud.githubusercontent.com/assets/1320353/7943038/1289c9b6-0930-11e5-830f-a43cff823231.png)
